elytron/WFCORE-5170-multiple-certificate-revocation-lists.adoc

```diff
@@ -0,0 +1,127 @@
+= [WFCORE-5170] Certificate Revocation Lists
+:author:            Sonia Zaldana Calles
+:email:             szaldana@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Certificate Revocation Lists (CRLs) contain a list of certificates that have been revoked
+by the issuing Certificate Authority before their expected expiration date, and therefore should no longer
+be trusted. Elytron currently supports configuring one Certificate Revocation List.
+
+However, if several Certificate Authorities are used, there is no way to configure several
+CRL files. This task is to enable support for more than one Certificate Revocation List in Elytron.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5170[WFCORE-5170]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-14007[WFLY-14007]
+* https://issues.redhat.com/browse/ELY-2033[ELY-2033]
+* https://issues.jboss.org/browse/EAP7-1548[EAP7-1548]
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+TBD
+
+=== Affected Projects or Components
+
+* WildFly, WildFly-Core and Elytron
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+* Add a new attribute ``certificate-revocation-lists`` in the ``trust-manager`` resource in the Elytron subsystem:
+
+** The ``trust-manager`` must be able to use the CRLs obtained from the distribution points referenced
+in the certificates using the ``certificate-revocation-lists`` attribute as follows:
+
+    /subsystem=elytron/trust-manager=myTrustManager:write-attribute(name=certificate-revocation-lists,value=[])
+
+** Alternatively, the ``trust-manager`` must be able to configure CRLs using certificate revocation
+list objects (which override the CRLs referenced in the distribution points). These objects have the
+attributes ``path`` and ``relative-to`` which specify the path to the CRL and the
+base path to the CRL respectively. These objects can be configured follows:
+
+    /subsystem=elytron/trust-manager=myTrustManager:write-attribute(name=certificate-revocation-lists, value=[{path=/path/to/CRL_FILE.crl.pem, relative-to=/SOME/DIRECTORY},
+    {path=/path/to/CRL_FILE2.crl.pem, relative-to=/SOME/DIRECTORY}])
+
+* The attribute ``certificate-revocation-list`` in the ``trust-manager`` resource
```

Comment:
I wonder can by `certificate-revocation-list` transformed into `certificate-revocation-lists` automatically by transformers? Is it good idea? (This is just question. I don't know what is best practice in subsystem development)

Reply:
Hi Martin, subsystem transformers usually transform new attributes to older ones for backwards compatibility and not the other way around. In previous similar cases, the approach taken was to simply deprecate the existing attribute. Please let me know if you have any other concerns.

Reply:
I think we will likely be better keeping the two attributes and making them mutually exclusive - the problem with the attributes that can take multiple values is the syntax in the CLI is more complex so I think retaining both makes sense.

Reply:
Sounds good. I'll update the proposal.

Reply:
+1, making them mutually exclusive makes sense.